pg_search/src/query/heap_field_filter.rs

```diff
@@ -114,27 +167,72 @@ impl HeapFieldFilter {
             return false;
         }
 
-        // Create an expression context for evaluation
-        let econtext = pg_sys::CreateStandaloneExprContext();
-        if econtext.is_null() {
-            pg_sys::ExecDropSingleTupleTableSlot(slot);
-            if buffer != pg_sys::InvalidBuffer as pg_sys::Buffer {
-                pg_sys::ReleaseBuffer(buffer);
+        // Use provided expression context if available, otherwise create a standalone one
+        let (econtext, should_free_context) = if !provided_expr_context.is_null() {
+            // Use the provided context (supports subqueries)
+            (provided_expr_context, false)
+        } else {
+            // Create a standalone context (fallback for simple expressions)
+            let standalone_context = pg_sys::CreateStandaloneExprContext();
```

Comment:
Could this happen in `fn evaluate`, so that there is one codepath which is either fetching the thread local or creating a standalone? You might also be able to create it using https://docs.rs/pgrx/latest/pgrx/pgbox/struct.PgBox.html to handle freeing it.

Reply:
I looked at it and it's not worth doing it, as we'd still need to do some other logic based on whether we're given a `expr_context` or not. Please look at the other usages of `expr_context` in this function.